src/ansys/geometry/core/designer/body.py

```diff
@@ -1763,6 +1803,38 @@ def plot(  # noqa: D102
         pl.plot(mesh_object, **plotting_options)
         pl.show(screenshot=screenshot, **plotting_options)
 
+    def export_gltf(  # noqa: D102
+        self,
+        merge: bool = False,
+        screenshot: str | None = None,
+        use_trame: bool | None = None,
+        use_service_colors: bool | None = None,
+        **plotting_options: dict | None,
+    ) -> None:
+        # lazy import here to improve initial module load time
+        import ansys.geometry.core as pyansys_geometry
+        from ansys.geometry.core.plotting import GeometryPlotter
+
+        use_service_colors = (
+            use_service_colors
+            if use_service_colors is not None
+            else pyansys_geometry.USE_SERVICE_COLORS
+        )
+
+        pl = GeometryPlotter(use_trame=use_trame, use_service_colors=use_service_colors)
+
+        faces = self.faces
+        dataset = self.tessellate()
+        for i, block in enumerate(dataset):
+            if faces[i].color != Color.DEFAULT.value:
+                plotting_options["color"] = faces[i].color
+            else:
+                plotting_options["color"] = self.color
+            pl._backend.pv_interface.plot(block, **plotting_options)
```

Comment:
This logic is going to have to be in the plotter now and it will have to assume:

If the face has a color --> retrieve that color
If it doesn't, use the parent's color (body)

Just like you did. But it should be implemented inside the plotter so that it is available for everybody. Not just for the export